main.py

```diff
@@ -1,83 +1,119 @@
+        resp = requests.put(
+            url=signed_url_info["url"],
+            data=data,
+            headers={"Content-Type": "application/octet-stream"}
+        )
+
         if resp.status_code != 200:
-            print(f"{resp.text} {resp.status_code}")
-            raise Exception('failed to upload file')
-    return signed_url_info['fileId']
+            print(f"Error code: {resp.status_code}", file=sys.stderr)
+            print(f"Error message: {resp.text}", file=sys.stderr)
+            raise Exception("failed to upload file")
+
+    return signed_url_info["fileId"]
+
 
 def upload_build(
-    input_file_id, # required
-    mapping_file_id, # optional
-    library_file_id, # optional
-    build_commit_hash, #optional
-    build_tags, # optional
-    ):
+    input_file_id,
+    mapping_file_id,
+    library_file_id,
+    commit_hash,
+    tags,
+    headers,
+    url
+):
     new_build_request = {
-        'inputFileId': input_file_id,
-        'mappingFileId': mapping_file_id,
-        'libraryFileId': library_file_id,
-        'commitHash': build_commit_hash,
-        'tags': build_tags,
-        'source': 'api'
+        "inputFileId":   input_file_id,
+        "mappingFileId": mapping_file_id,
+        "libraryFileId": library_file_id,
+        "commitHash":    commit_hash,
+        "tags":          tags,
+        "source":        "api"
     }
-    resp = requests.post(f"{url}/api/v0/builds", json=new_build_request, headers=HEADERS)
+
+    resp = requests.post(
+        url=f"{url}/api/v0/builds",
+        json=new_build_request,
+        headers=headers
+    )
+
     if resp.status_code != 200:
-        print(f"{resp.text} {resp.status_code}")
-        raise Exception('failed to create build')
+        print(f"Error code: {resp.status_code}", file=sys.stderr)
+        print(f"Error message: {resp.text}", file=sys.stderr)
+        raise Exception("failed to create build")
 
-    build_url = resp.json()['details']['buildUrl']
+    build_url = resp.json()["details"]["buildUrl"]
     print(f"Created a new build at: {build_url}")
 
-if __name__ == "__main__":
-    input_file_id = upload_file(inputFile)
-    mapping_file_id = upload_file(mappingFile)
-    library_file_id = upload_file(libraryFile)
+
+def main():
+    # default URL for AppSweep
+    url = "https://appsweep.guardsquare.com"
+    # Your AppSweep Project API key (stored as a SECRET)
+    api_key = os.environ["APPSWEEP_API_KEY"]
+    # Path to your APK file to upload and scan
+    input_file = os.environ["INPUT_FILE"]
+    # The commit hash to use for the build
+    commit_hash = os.environ["COMMIT_HASH"]
+    # The obfuscation mapping file of the build [Optional]
+    mapping_file = os.environ.get("MAPPING_FILE", None)
+    # The library mapping file of the build [Optional]
+    library_file = os.environ.get("LIBRARY_FILE", None)
+    # The tags to append to the build [Optional]
+    tags = os.environ.get("TAGS", None)
+
+    if tags is not None:
```

Comment:
Do we explain this expected format for the tags somewhere? I wasn't able to find it, and also I think this could have problems if people put spaces in the tags, so e.g. "release, othertag" -> "release", " othertag"

Reply:
All tags have been striped to make sure there is no space in them.